src/manifest/mod.ts

```diff
@@ -32,6 +32,18 @@ export const Manifest = (
   return manifest.export();
 };
 
+export const validateOutgoingDomains = (
+  domains: string[],
+) => {
+  return domains.map((domain) => {
+    try {
+      return new URL(domain).hostname;
+    } catch (e) {
+      throw new Error(`Invalid outgoing domain: ${domain}, error ${e}`);
```

Comment:
That's a good point about breaking change, but I think it may be warranted here. The current behaviour is: any string is accepted, even non-hostnames. That's not great as it may lead the developer into a false sense of security if they e.g. typo'ed a domain.

We may also want to raise this with the backend team to see if the manifest API should raise an error if the strings provided to `outgoingDomains` are not a hostname.

Reply:
Good point about raising this with the backend team. It would be more reliable and consistent to have the backend validate the `outgoingDomains` format.

Reply:
Glad these concerns are being discussed, thanks for raising them! I implemented @WilliamBergamin's suggestion above, but have implemented another check where the http protocol is assumed in the absence of the protocol, and then the hostname is returned.

If that constructor fails for some reason, there is another defensive check that validates if the domain is valid and returns it, or throws an error: https://github.com/slackapi/deno-slack-sdk/pull/246/commits/b584112331dcfa2a2b1dd4b42f0b192741024740

https://github.com/slackapi/deno-slack-sdk/blob/b584112331dcfa2a2b1dd4b42f0b192741024740/src/manifest/mod.ts#L35-L57

Is this the direction the team would like to take, or prefer having these checks in the backend side of things and not the SDK? I am leaning towards the latter. If agreed, I'll limit the changes to William's suggestion.

Reply:
One concern that still remains is, local hostnames like `localhost` are considered valid in the current implementation (without this PR's changes as well). Is that expected?

I'd imagine only fully qualified external domains are valid based on this documentation:

https://api.slack.com/automation/admin

>Outgoing domains are a new concept, and apply only to apps deployed to Slack's managed infrastructure. These are domains the app may require access to — for example, if a developer writes a [function](https://api.slack.com/automation/functions) that makes a request to an external API, they will need to include that API in their outgoing domains.

Reply:
This is great @arunsathiya ! We can pursue both in parallel; no reason why the SDK can't be improved while we're here, and separately we can work on improving the backend internally, too.

Reply:
Here are some test cases for the `isValidDomain` regex:

https://regexr.com/7nknp

Reply:
I wouldn't worry about validating the domain. I think we'd want just want to make sure we can support named servers and IP addresses - at least IPv4. The outgoing domains field gets passed through directly to `deno run --allow-net` (see [this code](https://github.com/slackapi/deno-slack-runtime/blob/main/src/local-run.ts#L68-L81)) - so whatever [the `--allow-net` deno flag supports](https://docs.deno.com/runtime/manual/basics/permissions#network-access), we'd want to make sure the Manifest supports it.